### PR TITLE
Added warning for future package name change

### PR DESCRIPTION
**Pull request by chris-theroux**
### Summary
This change adds a future warning for the upcoming package name change to qiskit-quantinuum-provider.

### CODE REVIEW

File: qiskit/providers/honeywell/__init__.py [hunk 1]
```diff
@@ -26,11 +26,18 @@
...
+import warnings
+
 from .version import __version__
 
 from .honeywellprovider import HoneywellProvider
 from .honeywellbackend import HoneywellBackend
 from .honeywelljob import HoneywellJob
 
+# Setting future warning for package name change
+warnings.warn("This is the final release of qiskit-honeywell-provider. " \
```
Comment (mtreinish) [inline qiskit/providers/honeywell/__init__.py]:
```suggestion
warnings.warn("This is the final release of qiskit-honeywell-provider. "
```

**Review by mtreinish — COMMENTED** [review]
The code LGTM except for the lint failure CI flagged. As for the test failure there was a importlib-metadata release today that broken a dependency of a dependency of the test runner used that's causing the failure. Let me push up a separate PR to pin that to unblock this.

File: qiskit/providers/honeywell/__init__.py (hunk 1 above)
Comment (chris-theroux) [reply qiskit/providers/honeywell/__init__.py]: Manually added this change

**Review by mtreinish — APPROVED** [review]
This LGTM, thanks for doing this, I'll tag and push the release tomorrow.

**Pull request merged**